### PR TITLE
Alarm on failed emails

### PR DESCRIPTION
**Pull request by mario-galic**
Related PRs: 

- https://github.com/guardian/support-service-lambdas/pull/635
- https://github.com/guardian/support-service-lambdas/pull/636

Fixes no alarms on any error.

### CODE REVIEW

**Comment by mario-galic** [comment]
[TASK: Fix silent failures in email pipeline](https://trello.com/c/Y9bZ97Mk/1108-task-fix-silent-failures-in-email-pipeline)

File: handlers/batch-email-sender/cfn.yaml [hunk 1]
```diff
@@ -52,6 +56,29 @@ Resources:
                   - "arn:aws:sqs:${AWS::Region}:${AWS::AccountId}:${QueueName}"
                   - { QueueName: !FindInMap [ StageMap, !Ref Stage, EmailSendingQueueName ]}
 
+  FailedRefundAlarm:
```
Comment (twrichards) [inline handlers/batch-email-sender/cfn.yaml:59]: Probably not the logical name you want

**Comment by twrichards** [comment]
This lambda has some fairly thorough documentation https://github.com/guardian/support-service-lambdas/blob/e9888f89adee375f85bfc10fec69f1ef36ea0916/handlers/batch-email-sender/README.md#responses , that specifically talks about the various response codes.

**Comment by mario-galic** [comment]
>  that specifically talks about the various response codes.

Those response code were never returned. All errors were previously swallowed and 200 was returned.

**Comment by mario-galic** [comment]
> I think there's deeper improvements that could be made here.

In #636 I introduced the following fixme:

```
FIXME: batch-email-sender seems to be implemented on partial success principle in which case what should the response code be?
```

Partial success is much harder to model. It means Salesforce would have to parse the response for both failures and success and have specific logic for those types of email that have to set the corresponding flag.

**Pull request merged**